tests/framework/extensions/provisioning/creates.go

```diff
@@ -227,22 +227,29 @@ func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvide
 		}
 	}
 
-	if clustersConfig.Hardened && clustersConfig.KubernetesVersion <= string(provisioninginput.HardenedKubeVersion) {
-		err = hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
-		}
-
-		hardenCluster := clusters.HardenK3SRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+	if clustersConfig.Hardened {
+		if strings.Contains(clustersConfig.KubernetesVersion, clusters.K3SClusterType.String()) {
+			err = k3sHardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode, clustersConfig.KubernetesVersion)
+			if err != nil {
+				return nil, err
+			}
 
-		_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
-		if err != nil {
-			return nil, err
-		}
+			hardenCluster := clusters.HardenK3SClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			err = rke2Hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
+			if err != nil {
+				return nil, err
+			}
 
-		err = hardening.PostHardeningConfig(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
+			hardenCluster := clusters.HardenRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
+			if err != nil {
+				return nil, err
+			}
 		}
```

Comment:
another really small nit; I think it would be worth adding a log that 'hardening is complete'

Reply:
Addressed